lib/ecto/uuid.ex

```diff
@@ -13,8 +13,14 @@ defmodule Ecto.UUID do
   @doc """
   Casts to UUID.
   """
-  def cast(<< _::64, ?-, _::32, ?-, _::32, ?-, _::32, ?-, _::96 >> = u), do: {:ok, u}
-  def cast(_), do: :error
+  def cast(uuid) do
+    regex = ~r/\A[a-f0-9]{8}-[a-f0-9]{4}-[1-5][a-f0-9]{3}-[89ab][a-f0-9]{3}-[a-f0-9]{12}\z/i
```

Comment:
Good point! I didn't quite realise it's possible, but now I see that this is what `dump` is doing. I don't really like regular expressions anyway :) The build is green now.